File: Python/array/get_equal_substrings_within_budget.py
def equalSubstring(s: str, t: str, maxCost: int) -> int:
    length = len(s)
    left, right, sum, result = 0, 0, 0, 0

    while right < length:
        sum += abs(ord(s[right]) - ord(t[right]))
        if sum <= maxCost:
            result = max(result, right - left + 1)
        else:
            sum -= abs(ord(s[left]) - ord(t[left]))
            left += 1
        right += 1
    
    return result
    
if __name__ == "__main__":
    print(equalSubstring(s = "krrgw", t = "zjxss", maxCost = 19))

File: src/tmp/Java/src/com/dbc/algorithm/Array/GetEqualSubstringsWithinBudget.java
package com.dbc.algorithm.Array;

public class GetEqualSubstringsWithinBudget {
    public static int equalSubstring(String s, String t, int maxCost) {
        int result = 0, left = 0, right = 0, sum = 0;
        while (right < s.length()) {
            sum += Math.abs(s.charAt(right) - t.charAt(right));
            if (sum <= maxCost) {
                result = Math.max(result, right - left + 1);
            } else {
                sum -= Math.abs(s.charAt(left) - t.charAt(left));
                left++;
            }
            right++;
        }
        return result;
    }
}
